android/app/src/main/java/com/microsoft/codepush/common/connection/QueryUpdateJob.java

```diff
@@ -0,0 +1,89 @@
+package com.microsoft.codepush.common.connection;
+
+import com.microsoft.appcenter.utils.AppCenterLog;
+import com.microsoft.codepush.common.datacontracts.CodePushUpdateResponse;
+import com.microsoft.codepush.common.exceptions.CodePushFinalizeException;
+import com.microsoft.codepush.common.exceptions.CodePushQueryUpdateException;
+import com.microsoft.codepush.common.utils.CodePushUtils;
+import com.microsoft.codepush.common.utils.FileUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.util.Arrays;
+import java.util.Scanner;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import static com.microsoft.codepush.common.CodePush.LOG_TAG;
+
+/**
+ * Performs sending status reports to server.
+ */
+public class QueryUpdateJob extends BaseJob<CodePushUpdateResponse> {
+
+    /**
+     * Url to query update against.
+     */
+    private String mRequestUrl;
+
+    /**
+     * Creates an instance of {@link QueryUpdateJob}.
+     *
+     * @param fileUtils     instance of {@link FileUtils} to work with.
+     * @param codePushUtils instance of {@link CodePushUtils} to work with.
+     */
+    public QueryUpdateJob(FileUtils fileUtils, CodePushUtils codePushUtils) {
+        mFileUtils = fileUtils;
+        mCodePushUtils = codePushUtils;
+    }
+
+    /**
+     * Sets additional parameters to the job.
+     *
+     * @param requestUrl url to query update against.
+     */
+    public void setParameters(String requestUrl) {
+        mRequestUrl = requestUrl;
+    }
+
+    @Override protected CodePushUpdateResponse doInBackground(Void... voids) {
+        InputStream stream = null;
+        Scanner scanner = null;
+        HttpURLConnection connection;
+        try {
+            connection = createConnection(mRequestUrl);
+        } catch (IOException e) {
+
+            /* We can't throw custom errors from this function, so any error will be passed to the result. */
+            return CodePushUpdateResponse.createFailed(new CodePushQueryUpdateException(e));
+        }
+        try {
```

Comment:
Do we really need to begin `try {` from this line? May be we should move it to 72?

Reply:
@ruslan-bikkinin Nope. The 62 line is already throwing an exception.

Reply:
@annakocheshkova ok.